frappe/model/document.py

```diff
@@ -328,6 +328,12 @@ def update_child_table(self, fieldname, df=None):
 				and parenttype=%s and parentfield=%s""".format(df.options),
 				(self.name, self.doctype, fieldname))
 
+	def get_doc_before_save(self):
+		self._doc_before_save = None
```

Comment:
the whole purpose is lost here! the goal was to no re-read if it was called again by another method.